compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -301,6 +301,9 @@ pub trait Machine<'tcx>: Sized {
     /// Determines the result of a `NullaryOp::ContractChecks` invocation.
     fn contract_checks(_ecx: &InterpCx<'tcx, Self>) -> InterpResult<'tcx, bool>;
 
+    /// Determines whether floating-point operations can behave non-deterministically.
+    fn float_nondet(_machine: &Self) -> bool;
```

Comment:
This is way too unspecific and also doesn't actually correctly describe how you use this function.

Please make this `fn float_fuse_mul_add` and move it to just after `equal_float_min_max`. Also the argument should be `_ecx: &InterpCx<'tcx, Self>`.